### DISCUSSION TITLE
umi4使用initialState插件报错

### DISCUSSION THREAD

**Discussion by liect** [umijs/umi#8288] [General]
# Umi4使用initialState插件报错
> 使用最新4.0.2版本,使用基础版本Umi,非Max,安装的是@umijs/plugins
### 配置文件如下
![截图_20220629095012](https://user-images.githubusercontent.com/16420004/176334377-bd26e8da-865d-47cd-b860-5316f050d15b.png)
### 启用initialState插件后报错如下,查看.umi目录确实没有生成插件
![微信截图_20220629094936](https://user-images.githubusercontent.com/16420004/176334605-72fcdb75-5af8-4dd2-ae22-8197b3adad80.png)

![微信截图_20220629094952](https://user-images.githubusercontent.com/16420004/176334619-21e51621-7887-49fb-9c31-ab8da2f43385.png)
### 只启用其他三个插件,.umi目录下会正常生成插件
![微信截图_20220629095208](https://user-images.githubusercontent.com/16420004/176334832-1163f5fc-065e-4c9d-92e6-e6aa577e495e.png)

**Comment by fz6m** [comment]
第一张图的两行注释（导入 plugin 和 `initialState: {}`）都放开才能使用 initialState 功能，`.umi/*` 里才会生成插件目录。

**Reply by liect** [reply]
我就是开启了,报的这个错误
![微信截图_20220629120931](https://user-images.githubusercontent.com/16420004/176349813-93493cdd-52b2-4ae9-8411-9b28e5eba6c6.png)

**Reply by fz6m** [reply]
将 `initial-state` 在 `plugins` 中的顺序移到第一位。

**Reply by liect** [reply]
> 将 `initial-state` 在 `plugins` 中的顺序移到第一位。
### 感谢大佬
成功了,按照逻辑应该model第一位,然后initial-state才对,属实没想到,感觉这个可以在官网提示一下

**Reply by xiaorong61** [reply]
我也这么觉得，const { initialState } = useModel('@@initialState') 看起来像 initialState 依赖 model
